### PR TITLE
feat: Add support for bulk update and delete

### PR DESCRIPTION
**Pull request by thewizkid87**

### CODE REVIEW

File: crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateTransactionalHandler.java [hunk 1]
```diff
@@ -17,4 +18,7 @@ default <ID extends Serializable, Entity extends BaseCrudEntity<ID>> Entity upda
 
 	<ID extends Serializable, Entity extends BaseCrudEntity<ID>> Entity updateTransactional(Entity entity, DynamicModelFilter filter, List<CRUDOnUpdateHook<ID, Entity>> onHooks, List<FieldChangeHook> fieldChangeHooks, boolean applyPolicies);
 	<ID extends Serializable, Entity extends BaseCrudEntity<ID>> Entity updateFromTransactional(DynamicModelFilter filter, Object object, Class<Entity> clazz, List<CRUDOnUpdateFromHook<ID, Entity>> onHooks, List<FieldChangeHook> fieldChangeHooks, boolean applyPolicies);
+
+    @Transactional(readOnly = false)
```
Comment (Idane) [inline crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateTransactionalHandler.java:22]: Remove from interface

File: crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateHandlerImpl.java [hunk 2]
```diff
@@ -151,5 +151,64 @@ public <ID extends Serializable, Entity extends BaseCrudEntity<ID>> Entity updat
...
+    @Override
+    public <ID extends Serializable, Entity extends BaseCrudEntity<ID>> List<Entity> bulkUpdate(List<Entity> entities, HooksDTO<CRUDPreUpdateHook<ID, Entity>, CRUDOnUpdateHook<ID, Entity>, CRUDPostUpdateHook<ID, Entity>> hooks, boolean applyPolicies) {
+        if (entities.isEmpty()) {
+            throw new IllegalStateException("Cannot update an empty list of entities");
+        }
+
+        Class<Entity> entityClazz = (Class<Entity>) entities.get(0).getClass();
+        List<FieldChangeHook> fieldChangeHooks = crudHelper.getFieldChangeHooks(entityClazz);
+        DynamicModelFilter filter = new DynamicModelFilter()
+                .add(FilterFields.in("id", FilterFieldDataType.get(entities.get(0).getId().getClass()), entities.stream().map(BaseCrudEntity::getId).toArray()));
```
Comment (Idane) [inline crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateHandlerImpl.java:163]: May need to box the type, see `CrudHelperImpl#getDataTypeFromClass`

**Review by Idane — DISMISSED** [review]

File: crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudDeleteHandlerImpl.java [hunk 3]
```diff
@@ -75,4 +77,54 @@ public <ID extends Serializable, Entity extends BaseCrudEntity<ID>> void deleteI
 			postHook.run(entity);
 		}
 	}
+
+    @Override
+    public <ID extends Serializable, Entity extends BaseCrudEntity<ID>> void bulkDelete(List<ID> ids, Class<Entity> entityClazz, HooksDTO<CRUDPreDeleteHook<ID, Entity>, CRUDOnDeleteHook<ID, Entity>, CRUDPostDeleteHook<ID, Entity>> hooks, boolean applyPolicies) {
+
+        Class<ID> idClass = (Class<ID>) ids.get(0).getClass();
```
Comment (Idane) [inline crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudDeleteHandlerImpl.java:84]: I think we should start relying on reflection less and EntityMetadataDTO more, meaning fetch metadata here and add if neeeded its ID class and ID name

Comment (Idane) [reply crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudDeleteHandlerImpl.java:84]: Already being fetched in this method so would make sense to move it up and add the additional info

File: crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateHandlerImpl.java [hunk 4]
```diff
@@ -151,5 +151,64 @@ public <ID extends Serializable, Entity extends BaseCrudEntity<ID>> Entity updat
 		return entity;
 	}
 
+    @Override
+    public <ID extends Serializable, Entity extends BaseCrudEntity<ID>> List<Entity> bulkUpdate(List<Entity> entities, HooksDTO<CRUDPreUpdateHook<ID, Entity>, CRUDOnUpdateHook<ID, Entity>, CRUDPostUpdateHook<ID, Entity>> hooks, boolean applyPolicies) {
+        if (entities.isEmpty()) {
+            throw new IllegalStateException("Cannot update an empty list of entities");
+        }
+
+        Class<Entity> entityClazz = (Class<Entity>) entities.get(0).getClass();
```
Comment (Idane) [inline crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateHandlerImpl.java:160]: See comment regarding EntityMetadataDTO

File: crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateHandlerImpl.java (hunk 2 above)
Comment (thewizkid87) [reply crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateHandlerImpl.java:163]: Moved the private into the enum, solves the TODO there as well

File: crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateHandlerImpl.java (hunk 4 above)
Comment (thewizkid87) [reply crud-framework-core/src/main/java/dev/krud/crudframework/crud/handler/CrudUpdateHandlerImpl.java:160]: I cant get the metadata without the entity class that i am getting from the list, its not reflection

**Pull request merged**